StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -2164,6 +2805,171 @@ private Message GetSortedSetRemoveRangeByScoreMessage(RedisKey key, double start
+                                + 1                         // The stream entry ID
+                                + includeMaxLen             // 2 or 0 (MAXLEN keyword & the count)
+                                + includeApproxLen;         // 1 or 0
+
+            var values = new RedisValue[totalLength];
+
+            var offset = 0;
+
+            values[offset++] = entryId;
+
+            if (maxLength.HasValue)
+            {
+                values[offset++] = StreamConstants.MaxLen;
+
+                if (useApproximateMaxLength)
+                {
+                    values[offset++] = StreamConstants.ApproximateMaxLen;
+                }
+
+                values[offset++] = maxLength.Value;
+            }
+
+            for (var i = 0; i < streamFields.Length; i++)
+            {
+                values[offset++] = streamFields[i].Name;
+                values[offset++] = streamFields[i].Value;
+            }
+
+            return Message.Create(Database, flags, RedisCommand.XADD, key, values);
+        }
+
+        private Message GetStreamClaimMessage(RedisKey key, RedisValue consumerGroup, RedisValue assignToConsumer, long minIdleTimeInMs, CommandFlags flags, string[] messageIds, bool returnJustIds = false)
+        {
+            if (messageIds == null || messageIds.Length == 0)
+            {
+                throw new ArgumentException("messageIds must contain at least 1 messageId.");
+            }
+
+            // XCLAIM <key> <group> <consumer> <min-idle-time> <ID-1> <ID-2> ... <ID-N>
+            var values = new RedisValue[4 + messageIds.Length + (returnJustIds ? 1 : 0)];
+
+            var offset = 0;
+
+            values[offset++] = key.AsRedisValue();
+            values[offset++] = consumerGroup;
+            values[offset++] = assignToConsumer;
+            values[offset++] = minIdleTimeInMs;
+
+            for (var i = 0; i < messageIds.Length; i++)
+            {
+                values[offset++] = messageIds[i];
+            }
+
+            if (returnJustIds)
+            {
+                values[offset++] = StreamConstants.JustId;
+            }
+
+            return Message.Create(Database, flags, RedisCommand.XCLAIM, values);
+        }
+
+        private Message GetStreamPendingMessagesMessage(RedisKey key, RedisValue groupName, RedisValue minId, RedisValue maxId, int count, RedisValue consumerName, CommandFlags flags)
+        {
+            // > XPENDING mystream mygroup - + 10 [consumer name]
+            // 1) 1) 1526569498055 - 0
+            //    2) "Bob"
+            //    3) (integer)74170458
+            //    4) (integer)1
+            // 2) 1) 1526569506935 - 0
+            //    2) "Bob"
+            //    3) (integer)74170458
+            //    4) (integer)1
+
+            // See https://redis.io/topics/streams-intro.
+
+            if (count <= 0)
+            {
+                throw new ArgumentOutOfRangeException("count must be greater than 0.");
```

Comment:
`throw new ArgumentOutOfRangeException(nameof(count), "count must be greater than 0.");`